src/app/common/dialogs/export-pdf-dialog/export-pdf-dialog.component.ts

```diff
@@ -120,225 +191,280 @@ export class ExportPdfDialog extends BaseDialog {
-				// image
-				const canvasWidth = 100;
-				const canvasHeight = 100;
-				const marginXImage = (pageWidth - canvasWidth) / 2;
-				let badge_img = new Image();
-				badge_img.src = badgeClass.image;
-				this.doc.addImage(badge_img, 'JPEG', marginXImage, yPos, canvasWidth, canvasHeight);
-
-				// title
-				yPos += canvasHeight;
-				this.doc.setFontSize(32);
-				this.doc.setFont('Helvetica', 'bold');
-				let title = this.doc.splitTextToSize(badgeClass.name, cutoff - this.doc.getTextWidth('...'));
-				let titlePadding = 0;
-				let maxTitleRows = 2;
-				if (title.length > maxTitleRows) {
-					title[maxTitleRows - 1] = title[maxTitleRows - 1] + '...';
-				} else if (title.length < maxTitleRows) {
-					titlePadding = (15 / 2) * (maxTitleRows - title.length);
-					yPos += titlePadding;
-				}
-				for (let i = 0; i < maxTitleRows; i = i + 1) {
-					if (title[i]) {
-						yPos += 15;
-						this.doc.text(title[i], pageWidth / 2, yPos, {
+				html2canvas(this.outputElement.nativeElement, {
+					backgroundColor: null,
+					scale: scaleFactor,
+				})
+					.then((canvas) => {
+						canvas.height = 100 * scaleFactor;
+						canvas.width = 100 * scaleFactor;
+						const image = this.generatePdfBackground(canvas);
+
+						const logoWidth = 50;
+						const logoHeight = 30;
+						const marginXImageLogo = 5;
+
+						let firstName = this.profile.firstName;
+						let lastName = this.profile.lastName;
+						firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);
+						lastName = lastName.charAt(0).toUpperCase() + lastName.slice(1);
+
+						// calculate competencies firs to know how many pages the PDF will have
+						// and to apply the background on each page
+						if (num_competencies > 0) {
+							const esco = competencies.some((c) => c.escoID);
+							const competenciesPerPage = 10;
+							this.doc.addPage();
+							this.doc.addImage(image, 'PNG', 0, 0, pageWidth, pageHeight, undefined, 'NONE');
+
+							//  OEB Logo
+							this.doc.addImage(oeb_logo, 'PNG', marginXImageLogo, yPos + 10, logoWidth, logoHeight);
+							this.doc.setDrawColor('#492E98');
+							this.doc.line(pageWidth / 2 - 50, 25, pageWidth / 2 + 100, 25);
+
+							if (esco) {
+								this.doc.textWithLink(
+									'* Kompetenz nach ESCO: https://esco.ec.europa.eu/de',
+									10,
+									pageHeight - 20,
+									{
+										url: 'https://esco.ec.europa.eu/de',
+									},
+								);
+							}
+
+							for (let i = 0; i < num_competencies; i++) {
+								if (i != 0 && i % competenciesPerPage === 0) {
+									this.doc.addPage();
+									this.doc.addImage(image, 'PNG', 0, 0, pageWidth, pageHeight, undefined, 'NONE');
+									//  OEB Logo
+									this.doc.addImage(
+										oeb_logo,
+										'PNG',
+										marginXImageLogo,
+										yPos + 10,
+										logoWidth,
+										logoHeight,
+									);
+									this.doc.setDrawColor('#492E98');
+									this.doc.line(pageWidth / 2 - 50, 25, pageWidth / 2 + 100, 25);
```

Comment:
There's a lot of "magic numbers" here, but I assume it would be hard to extract them into meaningful constants or something like that?

Reply:
not sure if i would be able to find a good name for everything